src/api/documentsApi.ts

```diff
@@ -0,0 +1,70 @@
+import { spWebContext } from "api/SPWebContext";
+import { useMutation, useQuery, useQueryClient } from "@tanstack/react-query";
+
+/**
+ * Gets all requests
+ */
+export const useDocuments = (requestId: number) => {
+  return useQuery({
+    queryKey: ["documents", requestId],
+    queryFn: () => getDocuments(requestId),
+  });
+};
+
+const getDocuments = async (requestId: number) => {
+  const path = "requests/" + requestId;
+  return spWebContext.web
+    .getFolderByServerRelativePath(path)
+    .files.select(
+      "Name",
+      "TimeLastModified",
+      "ServerRelativeUrl",
+      "ModifiedBy",
+      "ModifiedBy/Id",
+      "ModifiedBy/EMail",
+      "ModifiedBy/Title",
+      "UniqueId"
+    )
+    .expand("ModifiedBy")
+    .orderBy("Name")<SPDocument[]>();
+};
+
+export const useDeleteDocument = (document: SPDocument) => {
+  const queryClient = useQueryClient();
+  return useMutation(
+    ["deleteDocument", document.Name],
+    async () => {
+      return spWebContext.web.getFileById(document.UniqueId).recycle();
+    },
+    {
+      onSuccess: async () => {
+        queryClient.invalidateQueries(["documents"]);
+      },
+    }
+  );
+};
+
+export const useAddDocument = (requestId: number) => {
```

Comment:
So I was looking at https://support.microsoft.com/en-au/office/restrictions-and-limitations-in-onedrive-and-sharepoint-64883a5d-228e-48f5-b3d2-eb39e07630fa to see if we need to add any restrictions for filenames, and suprisingly we now support # and % in SharePoint! I tested it and it worked with 1 exception... The DocumentView Download is failing if the filename has either of these characters in it.  The other restricted characters are all restricted by Windows so we should be good there.  Not sure if it is worth looking at Mac/iDevice to ensure they can't use a character SharePoint disallows?